plugins/bulk_upload/xml/batch/BulkUploadEngineXml.php

```diff
@@ -1097,7 +1103,7 @@ protected function sendItemUpdateData($entryId, $entry, array $flavorAssets, arr
 				KBatchBase::$kClient->thumbAsset->setContent($existingthumbAssets[$thumbParamsId], $thumbAssetsResource);
 			}
 			if (strpos($thumbAssetsResource->tags, self::DEFAULT_THUMB_TAG) !== false)
-				KBatchBase::$kClient->thumbAsset->setAsDefault($thumAsset->id);
+				KBatchBase::$kClient->thumbAsset->setAsDefault($thumbsAsset->id);
```

Comment:
Won't this throw an exception in case we failed to add/update in that case thumbsAsset wont have the id attribute set